TestClasses/ReadWriteTest.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env python3
+import sys
+import yaml
+from BaseTest import BaseTest
+from PerformanceTest import PerformanceTest
+import subprocess
+import os
+import time
+import numpy as np
+from zlib import adler32
+
+class ReadWriteTest(BaseTest, PerformanceTest):
+    def __init__(self, configFile=None, singleEndPoint=None, singleProt:list=None):
+        super().__init__()
+        self.groupedOutput = []
+        self.checksums = []
+
+        if configFile is not None:
+            with open(configFile) as config:
+                try:
+                    sites = yaml.safe_load(config)
+                    self.sitesList = sites['SITES']
+                except yaml.YAMLError as exc:
+                    print(exc)
+        
+        if singleEndPoint is not None:
+            self.sitesList = singleEndPoint
+        if singleProt is not None:
+            self.protocols = singleProt
+        #sys.exit()
+
+    
+    def subprocess(self, action:str, sourcePath=None, destinBaseNm=None, *args, timeout=None):
+        self.setup(action, sourcePath, destinBaseNm, *args)
+        self.groupedOutput = [] # May not need this to be an instance/class variable
+        self.checksums = []
+
+        #Read-write tests will use xrdcp and gfal-copy
+        #If the protocol is root, use xrootd. Otherwise the protocol we use is gfal-copy
+        #create a combination of each protocol, server and root
+
+        if action == 'load': self.sitesList = self.redirect
```

Comment:
is the following code (line43+) only meant to run for action load? if yes the indentation need to reflect that, otherwise returning after the needed code has run is good practice